mesonbuild/dependencies/base.py

```diff
@@ -753,15 +755,31 @@ def _search_libs(self, out, out_raw):
                 continue
             if arg.startswith('-L') and arg[2:] not in prefix_libpaths:
                 system_libpaths.add(arg[2:])
+        # collect all secondary library paths
+        secondary_libpaths = OrderedSet()
+        all_args = self._convert_mingw_paths(shlex.split(out_all))
+        for arg in all_args:
+            if arg.startswith('-L') and not arg.startswith(('-L-l', '-L-L')):
+                path = arg[2:]
+                if not os.path.isabs(path):
+                    # Resolve the path as a compiler in the build directory would
+                    path = os.path.join(self.env.get_build_dir(), path)
+                if path not in prefix_libpaths and path not in system_libpaths:
+                    secondary_libpaths.add(path)
+
         # Use this re-ordered path list for library resolution
         libpaths = list(prefix_libpaths) + list(system_libpaths)
         # Track -lfoo libraries to avoid duplicate work
         libs_found = OrderedSet()
         # Track not-found libraries to know whether to add library paths
         libs_notfound = []
         libtype = LibType.STATIC if self.static else LibType.PREFER_SHARED
-        # Generate link arguments for this library
+        # Generate link arguments for this library, by
+        # first appending secondary link arguments for ld
         link_args = []
+        if self.clib_compiler and get_compiler_is_linuxlike(self.clib_compiler):
```

Comment:
lets not use `get_compiler_is_linuxlike` please. I'm trying to delete that. Something like:
```python
if self.clib_compiler and self.clib.compiler.linker and isinstance(self.clib_compiler.linker, linkers.GnuLikeDynamicLinkerMixin):
```

Reply:
`linkers.GnuLikeDynamicLinkerMixin` exists nowhere in the Meson codebase

Reply:
you need to rebase, it's in mesonbuild/linkers.py